### ISSUE TITLE
Built-in browser incompatible with freetype2 2.11.0

### ISSUE THREAD

**Issue by kattjevfel** [ValveSoftware/steam-for-linux#7935]
#### Your system information

* Steam client version (build number or date): 1626824053
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:

After upgrading freetype2 2.10.4 to 2.11.0 I cannot access the store, library or game properties and is instead greeted with a black window. I get the following error in the console:

```
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
```
I also tried latest git master of freetype2 but I still get the same error.

Downgrading to freetype2 2.10.4 works, but is obviously not preferred. [Full log](https://gist.github.com/4ff5306509f1fd596f5f97f579ed4058)

#### Steps for reproducing this issue:

1. Install freetype 2.11.0 or higher
2. Launch Steam and try access the store/library/game properties
3. Get a black window and `./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language` in console.

**Comment by NTMan** [comment]
Please add Fedora distro as affected by this bug.
https://bugzilla.rpmfusion.org/show_bug.cgi?id=6053

![Screenshot from 2021-07-23 23-04-36](https://user-images.githubusercontent.com/200750/126824413-4ea009dc-8c9e-494f-81ab-9696ab6589ec.png)

**Comment by plasticbomb1986** [comment]
can confirm. After issuing sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
steam works again.

**Comment by catrielmuller** [comment]
Confirmed that work again using freetype2-2.10.4-1

**Comment by BlueDragon2232** [comment]
Confirmative commander!

**Comment by plasticbomb1986** [comment]
steam-runtime
Running Steam on arch rolling 64-bit
STEAM_RUNTIME is enabled automatically
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/attiladeak/.local/share/Steam/ubuntu12_32/steam
[2021-08-01 12:38:35] Startup - updater built Jul 30 2021 21:22:59
[2021-08-01 12:38:35] Opted in to client beta 'publicbeta' via beta file
You are in the 'publicbeta' client beta.
Installing breakpad exception handler for appid(steam)/version(1627685077)
[2021-08-01 12:38:35] Loading cached metrics from disk (/home/attiladeak/.local/share/Steam/package/steam_client_metrics.bin)
[2021-08-01 12:38:35] Using the following download hosts for Public, Realm steamglobal
[2021-08-01 12:38:35] 1. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-08-01 12:38:35] 2. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-08-01 12:38:35] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1627685077)
[2021-08-01 12:38:35] Verifying installation...
[2021-08-01 12:38:35] Verification complete
Loaded SDL version 2.0.15-6693122

(steam:6440): Gtk-WARNING **: 12:38:35.794: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Adwaita-dark/gtk-2.0/main.rc:733: error: unexpected identifier 'direction', expected character '}'

(steam:6440): Gtk-WARNING **: 12:38:35.795: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Adwaita-dark/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
Installing breakpad exception handler for appid(steam)/version(1627685077)
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
CAppInfoCacheReadFromDiskThread took 114 milliseconds to initialize
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Proceed to auto login
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Opted-in Controller Mask for AppId 0: 0

(steam:6440): Gtk-WARNING **: 12:38:36.727: gtk_disable_setlocale() must be called before gtk_init()
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
roaming config store loaded successfully - 5190 bytes.
migrating temporary roaming config store
BRefreshApplicationsInLibrary 1: 9ms

(steam:6440): Gtk-WARNING **: 12:38:37.633: gtk_disable_setlocale() must be called before gtk_init()

(steam:6440): GLib-CRITICAL **: 12:38:37.633: unquote_string_inplace: assertion 'err == NULL || *err == NULL' failed

(steam:6440): Gtk-WARNING **: 12:38:37.633: gtk_disable_setlocale() must be called before gtk_init()

(steam:6440): GLib-CRITICAL **: 12:38:37.633: unquote_string_inplace: assertion 'err == NULL || *err == NULL' failed

(steam:6440): Gtk-WARNING **: 12:38:37.633: gtk_disable_setlocale() must be called before gtk_init()

(steam:6440): GLib-CRITICAL **: 12:38:37.633: unquote_string_inplace: assertion 'err == NULL || *err == NULL' failed
Installing breakpad exception handler for appid(steam)/version(1627685077)
Failed to init SteamVR because it isn't installed
ExecCommandLine: "'/home/attiladeak/.local/share/Steam/ubuntu12_32/steam'"
Installing breakpad exception handler for appid(steam)/version(1627685077)
System startup time: 2.31 seconds
Fossilize INFO: Setting autogroup scheduling.
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
Installing breakpad exception handler for appid(steam)/version(1627685077)
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Fossilize INFO: Setting autogroup scheduling.
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
Fossilize INFO: Setting autogroup scheduling.
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
./steamwebhelper: symbol lookup error: /usr/lib/libfreetype.so.6: undefined symbol: hb_ot_tags_from_script_and_language

**Comment by Crisp-Casper** [comment]
> steam-runtime

Turn off testing branches and update (downgrade) to the stable versions and compatibility should be fine

**Comment by heftig** [comment]
I suspect this is the fault of steam shipping a libharfbuzz that is too old. `hb_ot_tags_from_script_and_language` was introduced in harfbuzz 2.0.0 in 2018, but the steam runtime contains harfbuzz 0.9.35.

Either steam needs to use system harfbuzz or its own freetype.

**Comment by kattjevfel** [comment]
> I suspect this is the fault of steam shipping a libharfbuzz that is too old. `hb_ot_tags_from_script_and_language` was introduced in harfbuzz 2.0.0 in 2018, but the steam runtime contains harfbuzz 0.9.35.
> 
> Either steam needs to use system harfbuzz or its own freetype.

Could've sworn I tried `steam-native` earlier but without it working. Either way it works now with steam-native from arch's repos.

**Comment by grazzolini** [comment]
steam-native was not affected for this, yes. Only steam-runtime is.

**Comment by Gotchfutchian** [comment]
Can confirm this also happens on Redcore Linux (Gentoo but your too lazy to install it). Ill see if there is a freetype2.10 version on Gentoo's or Redcore's packages.

**Comment by cece140607** [comment]
> can confirm. After issuing sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
> steam works again.

It works. Thanks bro.

**Comment by mjevans** [comment]
Summary post to keep the most recent reply focused on resolving the root cause of the issue, with a mention of the workarounds to try avoiding more "worked for me too" posts about those band-aids.

Quoting https://github.com/ValveSoftware/steam-for-linux/issues/7935#issuecomment-890558546 from ~ Aug 1, 2021

> I suspect this is the fault of steam shipping a libharfbuzz that is too old. `hb_ot_tags_from_script_and_language` was introduced in harfbuzz 2.0.0 in 2018, but the steam runtime contains harfbuzz 0.9.35.
> 
> Either steam needs to use system harfbuzz or its own freetype.

Workarounds until Valve fixes the Steam client:

* steam-native (force steam to use native harfbuzz)
* Downgrade local Freetype, which is unlikely to be viable and may include security vulnerabilities. (See prior posts for versions on some distributions.)

The upstream fix for steam-runtime will most likely take one of two paths, as suggested by heftig:

* use system harfbuzz and freetype
* ship it's own freetype in addition to it's own harfbuzz; more bits in the steam library collection (/snap/docker image/dll hell/etc)

**Comment by heftig** [comment]
I forgot the third option of upgrading the runtime's harfbuzz.

**Comment by Ejdesgaard** [comment]
> Yes; that was my reading also I just couldn't find the github issue to link. No ; whilst I could grab an out of tree older build the current version of freetype in rawhide/fc35 is targeted as : 2.11.0-1

Got same problem on F35-devel.
Confirmed that manually downgrading to F34 rpm's will also make steam work on F35-devel. (have not tested if it negatively affect other parts of F35-devel)
https://koji.fedoraproject.org/koji/buildinfo?buildID=1703857
1. Downloaded the x86_64 and i686 packages, including -devel
2. cd [download location]
3. `sudo dnf downgrade freetype-*`

**Comment by doogie544** [comment]
I can confirm that this is an issue in Arch as well if running freetype2 2.11.0-3.

Downgrading to freetype2-2.10.4-1 solves the issue


`Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  AuthenticAMD
    CPU Brand:  AMD Ryzen 7 3800X 8-Core Processor             
    CPU Family:  0x17
    CPU Model:  0x71
    CPU Stepping:  0x0
    CPU Type:  0x0
    Speed:  4200 Mhz
    16 logical processors
    8 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    AVX2:  Unsupported
    AVX512F:  Unsupported
    AVX512PF:  Unsupported
    AVX512ER:  Unsupported
    AVX512CD:  Unsupported
    AVX512VNNI:  Unsupported
    SHA:  Unsupported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    "Arch Linux" (64 bit)
    Kernel Name:  Linux
    Kernel Version:  5.13.13-arch1-1
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12013000
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime_0.20210630.0

Video Card:
    Driver:  NVIDIA Corporation NVIDIA GeForce RTX 2070 SUPER/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 470.63.01
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 144 Hz
    VendorID:  0x10de
    DeviceID:  0x1e84
    Revision Not Detected
    Number of Monitors:  2
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 3840 x 1080
    Primary Display Size: 20.91" x 11.77" (23.98" diag)
                                            53.1cm x 29.9cm (60.9cm diag)
    Primary Bus: PCI Express 16x
    Primary VRAM: 8192 MB
    Supported MSAA Modes:  2x 4x 8x 16x 

Sound card:
    Audio device: Nvidia GPU 92 HDMI/DP

Memory:
    RAM:  32011 Mb

VR Hardware:
    VR Headset: None detected

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  1876255 Mb
    Largest Free Hard Disk Block:  675601 Mb
`


(I can include the rest of my system info if needed, I truncated it as I suspect most of it is not relevant to this issue)

**Comment by doogie544** [comment]
I can also confirm that the latest Steam beta released today (version 163453407) fixed the issue with freetype2 2.11.0-3 installed

**Comment by aenertia** [comment]
Fixed on Fedora Rawhide as of Beta today.

**Comment by dluzak** [comment]
On Arch on Beta build it works as well.

**Comment by kattjevfel** [comment]
Had to opt-out and back in to get the new update, but can also confirm it works on arch with freetype2 2.11.0. Feel free to close this issue when it hits steam stable.

**Comment by eNTi** [comment]
Can confirm that the beta fixed the issue on Artix Linux.

**Comment by jatin-cbs** [comment]
I have posted the fix on Arch Linux reddit, if someone hasn't opted to steam beta [Link](https://www.reddit.com/r/archlinux/comments/pfovqy/note_to_steam_users_on_arch_linux/ )

**Comment by axredneck** [comment]
For me beta is affected by this issue but stable isn't for some reason. (But stable has other issues...)
Arch Linux.

**Comment by axredneck** [comment]
> Please, rebuild your pins and try again.

How can i do it?

**Comment by grazzolini** [comment]
> > Please, rebuild your pins and try again.
> 
> How can i do it?

Start with:

steam-runtime --reset

**Comment by Chryseus** [comment]
Had this issue pop up today (1 Sep) on Arch, I was running freetype 2.11.0-3 the day before without any issues.
Switching to steam beta solves the problem.

**Comment by axredneck** [comment]
@grazzolini, doesn't help, still black window. I'm on beta.

**Comment by 414n** [comment]
Now I get black screens with the latest beta client (just installed it 10 mins ago).
Didn't have the issue before and did not upgrade any system package in the meantime.
I see a bunch of `src/vgui2/vgui_surfacelib/FontManager.cpp (287) : Assertion Failed: descs.Count() >= 1` messages on stdout when launching Steam in terminal, but nothing else of interest.
System freetype library is 2.6.3 (Slackware64 14.2).

**Comment by kisak-valve** [comment]
Hello @414n, your issue doesn't sound related to this issue report. Please open a new issue report.

**Comment by 414n** [comment]
Hello @kisak-valve,

sorry, will open a separate issue then.

**Comment by sheepsushis** [comment]
> can confirm. After issuing sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
> steam works again.

can confirm, works after downgrade.

**Comment by AlexFolland** [comment]
This is working on Manjaro on the unstable branch with the latest package updates as well.  I did not have to reboot after opting into the Steam beta for the fix to work well.

**Comment by AlexFolland** [comment]
There should not be comments about downgrading Freetype, considering the issue is fixed while using the latest Freetype when using the latest Steam beta client.  Let's acknowledge that the bug has been fixed in Steam and not downgrade our packages.

**Comment by viriuwu** [comment]
Confirming the beta fixes it on Arch Linux (2021-09-02). Thank you.

**Comment by sheepsushis** [comment]
Confirming steam beta fixes the black screen issue on Arch Linux 5.13.13.arch1-1 using freetype2 2.11.0-3

**Comment by nmenetrier** [comment]
Problem fixed on Arch with freetype2-2.11.0-3 and steam beta, thanks :)

**Comment by ashtonx** [comment]
I can also confirm beta fixed the issue on arch with extra/freetype2 2.11.0-3.

**Comment by karlovskiy** [comment]
Confirming that last beta fixes black screen issue (freetype2 2.11.0-3)

**Comment by sgtnasty** [comment]
I can also confirm beta fixes issue.
```sh
System:    Host: thinkpad Kernel: 5.13.13-arch1-1 x86_64 bits: 64 Desktop: KDE Plasma 5.22.5 Distro: Arch Linux 
CPU:       Info: 6-Core model: Intel Core i7-8850H bits: 64 type: MT MCP cache: L2: 9 MiB 
Graphics:  Device-1: NVIDIA GP107M [GeForce GTX 1050 Ti Mobile] driver: nvidia v: 470.63.01 
Memory:    8567MiB / 31600MiB
```

**Comment by neetism** [comment]
> can confirm. After issuing sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst

>package missing required signature

I know I have to enable something in /etc/pacman.conf I just don't know what.

**Comment by doogie544** [comment]
> > can confirm. After issuing sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
> 
> "package missing required signature"
> 
> I know I have to enable something in /etc/pacman.conf I just don't know what.
> 
> Edit: Figure it out, set "SigLevel = Optional DatabaseOptional"
> More infonmation here: https://archlinux.org/pacman/pacman.conf.5.html


You would need to download the signature or downgrade to the one still in your cache.

However you can JUST USE THE STEAM BETA  instead as a better temporary workaround to putting your system in a partial upgrade state

**Comment by hafizdkren** [comment]
Approving EOS | BSPWM work on steam-runtime beta option. It mean it's not your tiling problem but more onto not able to finding the font steam client needed.
steam-runtime stable work on EOS | (adding) Openbox and showing everything on it. maybe because how WM and DE works so it can be say that need to workaround on steam client to show the client. (This is my opinion on how this problem exist, since I try openbox install and it works all the sudden where it show the browser while on bspwm it just black)

My better suggestion out there for non-beta is to installing additional steam-native for a moment, where you can download and open native game and also setting the game on tray just by open them.

Hence, you can get a best for both world, updated freetype and working steam. (For a moments)

**Comment by robbintt** [comment]
I'm having the black screen on beta issue in Ubuntu 20.10, did not use beta before today. Stable has no issues.

**Comment by kisak-valve** [comment]
Hello @robbintt, your issue is most likely unrelated to this issue report. Please open a new issue report.

**Comment by MyCookie** [comment]
IMO, a better, temporary solution was pointed out on the Arch subreddit: `LD_PRELOAD=/lib/libharfbuzz.so steam`. Please, no downgrading packages, it's very easy to forget to remove the upgrade lock once the fix is in stable.

**Comment by grazzolini** [comment]
> IMO, a better, temporary solution was pointed out on the Arch subreddit: `LD_PRELOAD=/lib/libharfbuzz.so steam`. Please, no downgrading packages, it's very easy to forget to remove the upgrade lock once the fix is in stable.

Well, you could also use steam-native. Also, if you downgrade some package, pacman will throw out a warning every time you run it to upgrade your packages, so, kind difficult to forget. Anyway, using steam-native, downgrading the freetype package, using steam beta, or the LD_PRELOAD method, all are valid solutions to this issue, until the fix comes out of beta.

**Comment by Dishoungh** [comment]
Arch user here: Can confirm the issue is still happening until I downgraded freetype2 package. Thank you, @plasticbomb1986

**Comment by AlexFolland** [comment]
> There should not be comments about downgrading Freetype, considering the issue is fixed while using the latest Freetype when using the latest Steam beta client. Let's acknowledge that the bug has been fixed in Steam and not downgrade our packages.

@Dishoungh Please see this comment.

**Comment by saltfishh** [comment]
> can confirm. After issuing sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
> steam works again.


thanks for that,steam do works again.
thanks!

**Comment by doogie544** [comment]
> > can confirm. After issuing sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
> > steam works again.
> 
> thanks for that,steam do works again.
> thanks!

There there are currently three options to get steam working again. While downgrading the package does work it is the least desirable fix. Packages should only be downgraded when there is absolutely no other choice. Before starting a debate as to which option is best and why you shouldn't downgrade I will state this. 

The bug report is requesting additional testing. This means using the steam beta client to see if the issue is fixed. Downgrading the package does not contribute to the requested testing so is of no help to getting the bug fixed. 

Please do as the bug assignee is requesting and test the steam beta client to get this fix released from beta as soon as possible.

**Comment by dhax** [comment]
another workaround instead of downgrading, at least fo rme on arch, is to switch to Big Picture Mode which still works

**Comment by doogie544** [comment]
> another workaround instead of downgrading, at least fo rme on arch, is to switch to Big Picture Mode which still works

I saw that mentioned someplace else and tested it just to see and while you can see the game so you can launch them most everything else is broken the storefront doesn't render if I recall correctly and I don't believe the friends list works the discussions don't work so it's only a partial work around

However we're still back to their requesting testing for the beta to make sure it's fixed properly. Using the regular client and Big picture mode is still not helping resolve the bug

**Comment by seyfahni** [comment]
> The bug report is requesting additional testing. This means using the steam beta client to see if the issue is fixed. Downgrading the package does not contribute to the requested testing so is of no help to getting the bug fixed. 
> 
> Please do as the bug assignee is requesting and test the steam beta client to get this fix released from beta as soon as possible. 

I can confirm that switching from Steam client release to beta solved the issue for me. I also verified that I'm using the current (with Steam client release incompatible) freetype2 version.

System: Arch Linux

**Comment by iamthewallrus** [comment]
> sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst

This worked for me. Thanks!

**Comment by jansalleine** [comment]
I can also confirm that switching to Steam beta client fixes the issue on my Arch Linux install.
```
[spider@misery ~]$ uname -a
Linux misery 5.13.13-arch1-1 #1 SMP PREEMPT Thu, 26 Aug 2021 19:14:36 +0000 x86_64 GNU/Linux
[spider@misery ~]$ pacman -Qs freetype
local/freetype2 2.11.0-3
    Font rasterization library
local/lib32-freetype2 2.11.0-3
    Font rasterization library (32-bit)
local/lib32-libxft 2.3.4-1
    FreeType-based font drawing library for X (32-bit)
local/libxft 2.3.4-1
    FreeType-based font drawing library for X
```

**Comment by qndydyhm** [comment]
switch to beta version do solve this problem on arch linux

**Comment by benneti** [comment]
I think the changes for this in the beta break steam on other distributions, i.e. on nixos stable works but on the current beta the store stays black and in 
.local/share/Steam/logs/steamwebhelper.log
```
Steam runtime environment up-to-date!
Warning: An unexpected error occurred while executing "/sbin/ldconfig -XNv", the exit status was 127
./steamwebhelper: /home/benneti/.local/share/Steam/ubuntu12_64/steam-runtime-heavy/usr/lib/x86_64-linux-gnu/libthai.so.0: version `LIBTHAI_0.1.25' not found (required by /lib64/libpango-1.0.so.0)
```

**Comment by creideiki** [comment]
Switching to the beta (build 1631058341) does not fix this problem for me on Gentoo with media-libs/freetype-2.11.0-r1. All Steam windows are still completely black, there are no `steamwebhelper` processes running, and the terminal where I started Steam outputs the following in an infinite loop:

```
Runtime for steamwebhelper: defaulting to /home/creideiki/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/creideiki/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/creideiki/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/creideiki/.local/share/Steam/config/htmlcache -steampid=4846 -buildid=1631058341 -steamid=76561197971747888 -cachedir=/home/creideiki/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/creideiki/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/creideiki/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
```

`steamwebhelper.log` is very brief and contains nothing useful:
```
Updating Steam runtime environment...
/home/creideiki/.local/share/Steam/ubuntu12_64/steam-runtime-heavy/setup.sh: line 421: zenity: command not found
/home/creideiki/.local/share/Steam/ubuntu12_64/steam-runtime-heavy/setup.sh: line 77: echo: write error: Broken pipe
```

**Comment by smcv** [comment]
@benneti:
> `Warning: An unexpected error occurred while executing "/sbin/ldconfig -XNv", the exit status was 127`

Please report this as a separate issue. It is likely to be specific to NixOS (and maybe other non-FHS distributions like Guix).

**Comment by smcv** [comment]
@creideiki:
> `/home/creideiki/.local/share/Steam/ubuntu12_64/steam-runtime-heavy/setup.sh: line 421: zenity: command not found`

This is ValveSoftware/steam-runtime#458. Workaround: install `zenity`. A fix for this is queued up for a future beta.

**Comment by creideiki** [comment]
> @creideiki:
> 
> > `/home/creideiki/.local/share/Steam/ubuntu12_64/steam-runtime-heavy/setup.sh: line 421: zenity: command not found`
> 
> This is [ValveSoftware/steam-runtime#458](https://github.com/ValveSoftware/steam-runtime/issues/458). Workaround: install `zenity`. A fix for this is queued up for a future beta.

Confirm. With Zenity installed, I can run the beta Steam client with Freetype 2.11.0.

**Comment by graves501** [comment]
> > sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
> 
> This worked for me. Thanks!

Both downgrading `freetype2` to version 2.10 and switching to the Steam Beta client with version 2.11 worked for me on Manjaro and Arch Linux.

**Comment by N0p1n** [comment]
Confirming this issue on manjaro with freetype2-2.11.0-3. 
Downgrading to 2.10.4-1 fixed this issue immediatly. 

Hints for pacman use:
1. Check local cache for old package for local downgrade
`ls -l /var/cache/pacman/pkg | grep freetype2-2.10.4-1-x86_64.pkg.tar.zst`
`sudo pacman -U /var/cache/pacman/pkg/freetype2-2.10.4-1-x86_64.pkg.tar.zst`
2. Ignore freetype2 until updated client
Add `IgnorePkg   = freetype2` to `/etc/pacman.conf`
OR
use `--ignore freetype2` when updating with pacman

**Comment by smcv** [comment]
Please don't downgrade freetype unless you absolutely have to. Using the beta is a better answer to this: it helps Valve (because they get reports of regressions like ValveSoftware/steam-runtime#458, and can fix them), and it helps you (because you aren't missing out on freetype updates that might be fixing a security vulnerability).

Sharing recipes for how to downgrade a package seems like it will encourage less experienced users to see downgrading as a desirable thing, and that's going to hurt someone sooner or later.

**Comment by tgharib** [comment]
It appears the built-in browser broke with the September 10 stable update on NixOS.

https://github.com/NixOS/nixpkgs/issues/137279

**Comment by kisak-valve** [comment]
Thanks everybody for their testing related to this issue report. Yesterday, the beta branch was promoted to mainline and troubleshooting on this issue report has concluded.

The solution to this issue did significantly change how the environment is setup for steamwebhelper and this revealed some other issues:

On systems without zenity installed, https://github.com/ValveSoftware/steam-runtime/issues/458 was a hidden issue which has now has more visible side effects.

Distributions without 64 bit libcups installed have encountered https://github.com/ValveSoftware/steam-for-linux/issues/8031.

Non-FHS compliant distro maintainers with a Steam package will need to take a look at their Steam packaging. For NixOS, this is also being tracked at https://github.com/ValveSoftware/steam-runtime/issues/462.

Each of these issues should be discussed separately, and while the high level symptoms look similar, they are unrelated to each other. Please do not use this issue report for near-match symptoms to the freetype / harfbuzz issue that was tracked here.

**Comment by gardotd426** [comment]
@kisak-valve, the issue has returned (at least on Arch Linux). I'm already participating in the Beta.

**Comment by kisak-valve** [comment]
Hello @gardotd426, please open a new issue report.

**Comment by jonaburg** [comment]
> > sudo pacman -U https://archive.archlinux.org/packages/path/freetype2-2.10.4-1-x86_64.pkg.tar.zst
> 
> This worked for me. Thanks!

This actually does not work for me on 5.13.13-arch1-1

**Comment by smcv** [comment]
> This actually does not work for me on 5.13.13-arch1-1

If you are still seeing this symptom, please open a **new** issue with as much information as possible.

If you can find an error message that did not previously appear, please mention it in the issue's title, to avoid people who have the same symptom for a different reason causing confusion and making a fix take longer.

We cannot solve "this does not work for me".

**Comment by mjevans** [comment]
My desktop is also Archlinux, I had been using steam-native as a workaround and just retested.  This works for me, at the present time.

Help > Steam Information

Operating System Version:
    "Arch Linux" (64 bit)
    Kernel Version:  5.13.13-arch1-1
    Steam Runtime Version:  steam-runtime_0.20210906.1

Jonaburg probably:

1) Should update to the current system packages rather than sticking to outdated packages and retest.
2) Should open a new issue if their symptoms persist.
3) It would be really great if Steam had an environment assumptions checker (even more ideal a command line script entry point for that documented); that data would be very useful when opening a new issue.

**Comment by smcv** [comment]
> It would be really great if Steam had an environment assumptions checker

This is what the blob of JSON in Help -> System Information is for. It comes from multiple sources, but `~/.steam/root/ubuntu12_32/steam-runtime/run.sh -- steam-runtime-system-info` is the main one.

Limitations: it only detects assumptions that we know about and are feasible to check (assumptions we don't know we have made are hard to check for!), and it doesn't currently cover the `heavy` mini-runtime used for the built-in browser (to do that, we'd have to make the `heavy` mini-runtime significantly larger, because the diagnostic tool itself has dependencies).

**Comment by fazo96** [comment]
SteamVR (specifically vrwebhelper.sh) is also affected by this issue, which remains unsolved as of now https://github.com/ValveSoftware/SteamVR-for-Linux/issues/465

**Comment by happysmash27** [comment]
Opting in to the Steam client update, finally, fixes [the issue with v2gui.so that requires me to delete a bunch of libraries every time](https://forums.gentoo.org/viewtopic-t-1110384.html); however, the black screen issue is unaffected, I am guessing because I never had any errors with missing libraries in the first place, as far as I can tell: 

```
Running Steam on gentoo 1.0.0.71 64-bit
STEAM_RUNTIME is enabled by the user
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/happysmash27/.local/share/Steam/ubuntu12_32/steam
[2021-10-03 13:39:31] Startup - updater built Sep 17 2021 04:27:28
Installing breakpad exception handler for appid(steam)/version(1631912732)
[2021-10-03 13:39:32] Loading cached metrics from disk (/home/happysmash27/.local/share/Steam/package/steam_client_metrics.bin)
[2021-10-03 13:39:32] Using the following download hosts for Public, Realm steamglobal
[2021-10-03 13:39:32] 1. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-10-03 13:39:32] 2. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-10-03 13:39:32] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1631912732)
[2021-10-03 13:39:32] Verifying installation...
[2021-10-03 13:39:32] Verification complete
Loaded SDL version 2.0.17-6744061
Gtk-Message: 13:39:33.993: Failed to load module "appmenu-gtk-module"
/usr/share/themes/Breeze-Dark/gtk-2.0/widgets/entry:70: error: unexpected identifier 'direction', expected character '}'

(steam:23586): Gtk-WARNING **: 13:39:34.155: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Breeze-Dark/gtk-2.0/widgets/styles:36: error: invalid string constant "combobox_entry", expected valid string constant
Installing breakpad exception handler for appid(steam)/version(1631912732)
Runtime for steamwebhelper: defaulting to /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/happysmash27/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steampid=23586 -buildid=1631912732 -steamid=0 -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/happysmash27/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/happysmash27/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
CAppInfoCacheReadFromDiskThread took 245 milliseconds to initialize
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Proceed to auto login
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Opted-in Controller Mask for AppId 0: 0

(steam:23586): Gtk-WARNING **: 13:39:45.013: gtk_disable_setlocale() must be called before gtk_init()
Runtime for steamwebhelper: defaulting to /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/happysmash27/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steampid=23586 -buildid=1631912732 -steamid=76561198153329814 -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/happysmash27/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/happysmash27/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
Runtime for steamwebhelper: defaulting to /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/happysmash27/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steampid=23586 -buildid=1631912732 -steamid=76561198153329814 -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/happysmash27/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/happysmash27/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
Installing breakpad exception handler for appid(steam)/version(1631912732)
roaming config store loaded successfully - 3194 bytes.
migrating temporary roaming config store
Installing breakpad exception handler for appid(steam)/version(1631912732)
BRefreshApplicationsInLibrary 1: 11ms
Installing breakpad exception handler for appid(steam)/version(1631912732)
Installing breakpad exception handler for appid(steam)/version(1631912732)
amdgpu: os_same_file_description couldn't determine if two DRM fds reference the same file description.
If they do, bad things may happen!
ExecCommandLine: "'/home/happysmash27/.local/share/Steam/ubuntu12_32/steam'"
System startup time: 33.28 seconds
Runtime for steamwebhelper: defaulting to /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/happysmash27/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steampid=23586 -buildid=1631912732 -steamid=76561198153329814 -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/happysmash27/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/happysmash27/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
Runtime for steamwebhelper: defaulting to /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/happysmash27/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steampid=23586 -buildid=1631912732 -steamid=76561198153329814 -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/happysmash27/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/happysmash27/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
Runtime for steamwebhelper: defaulting to /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/happysmash27/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steampid=23586 -buildid=1631912732 -steamid=76561198153329814 -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/happysmash27/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/happysmash27/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
Runtime for steamwebhelper: defaulting to /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
+ exec /home/happysmash27/.local/share/Steam/ubuntu12_64/steam-runtime-heavy.sh --unpack-dir=/home/happysmash27/.local/share/Steam/ubuntu12_64 --runtime=steam-runtime-heavy -- ./steamwebhelper -lang=en_US -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steampid=23586 -buildid=1631912732 -steamid=76561198153329814 -cachedir=/home/happysmash27/.local/share/Steam/config/htmlcache -steamuniverse=Public -realm=Global -clientui=/home/happysmash27/.local/share/Steam/clientui --enable-blink-features=ResizeObserver,Worklet,AudioWorklet --enable-media-stream --enable-smooth-scrolling --password-store=basic --log-file=/home/happysmash27/.local/share/Steam/logs/cef_log.txt --disable-quick-menu
saving roaming config store to 'sharedconfig.vdf'
roaming config store 2 saved successfully
Exiting workitem thread

(steam:23586): GLib-GObject-CRITICAL **: 13:40:44.043: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(steam:23586): GLib-GObject-WARNING **: 13:40:44.044: invalid (NULL) pointer instance

(steam:23586): GLib-GObject-CRITICAL **: 13:40:44.044: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(steam:23586): GLib-GObject-CRITICAL **: 13:40:44.044: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
[2021-10-03 13:40:44] Shutdown
```

May put this in as another issue if downgrading freetype does not help.

**Comment by kisak-valve** [comment]
Hello @happysmash27 the investigation on this issue report has concluded and your issue doesn't appear to be related. Please open a new issue report.

As a side note, please use a [gist](https://gist.github.com) or other pastebin service for logs instead of posting logs in-line so that it's easier to have a discussion along with the logs.

**Comment by happysmash27** [comment]
I have opened a new issue for this shortly after posting: https://github.com/ValveSoftware/steam-for-linux/issues/8123.

**Comment by codeandcreate** [comment]
Same here on debian/flathub version

**Comment by smcv** [comment]
> Same here on debian/flathub version

@codeandcreate: The issue that was tracked here has been fixed. If you are experiencing similar symptoms, please open a separate issue report with full details, and be as specific about it as you can, including any error messages that appear when Steam is run from a terminal.
